packages/cactus-plugin-cc-tx-visualization/src/main/typescript/plugin-cc-tx-visualization.ts

```diff
@@ -0,0 +1,580 @@
+      },
+    });
+  }
+
+  public watchRunTxReqWithTxId(): void {
+    const fnTag = `${this.className}#watchRunTxReqWithTxId()`;
+    this.log.debug(fnTag);
+
+    this.fabricTxObservable.subscribe({
+      next: (data: RunTxReqWithTxId) => {
+        // Handle the data received by the observer
+        this.pollTxReceiptsFabric(data); //trigger whenever a new value is observed
+      },
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      error: (error: any) => {
+        // Handle errors if any
+        console.error("Error:", error);
+      },
+    });
+  }
+
+  public monitorTransactions(): void {
+    const fnTag = `${this.className}#monitorTransactions()`;
+    this.log.debug(fnTag);
+
+    this.watchRunTxReqWithTxId();
+    this.watchRunTransactionV1Exchange();
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private createCrossChainEventFromBesuReceipt(besuReceipt: any): void {
+    const ccEventFromBesu: CrossChainEvent = {
+      caseID: besuReceipt.caseID,
+      receiptID: besuReceipt.transactionID,
+      blockchainID: besuReceipt.blockchainID,
+      invocationType: besuReceipt.invocationType,
+      methodName: besuReceipt.methodName,
+      parameters: besuReceipt.parameters,
+      timestamp: besuReceipt.timestamp.toISOString(),
+      identity: besuReceipt.from,
+      cost: gweiToDollar(calculateGasPriceBesu(besuReceipt.gasUsed as number)),
+      carbonFootprint: CarbonFootPrintConstants(besuReceipt.blockchainID),
+      latency: millisecondsLatency(besuReceipt.timestamp),
+      revenue: besuReceipt.revenue || 0,
+    };
+
+    this.crossChainLog.addCrossChainEvent(ccEventFromBesu);
+    this.log.info("Added Cross Chain event from BESU");
+    this.log.debug(`Cross-chain log: ${JSON.stringify(ccEventFromBesu)}`);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private createCrossChainEventFromFabricReceipt(fabricReceipt: any): void {
+    const ccEventFromFabric: CrossChainEvent = {
+      caseID: fabricReceipt.caseID,
+      receiptID: fabricReceipt.transactionID || `FABRIC-CALL-${randomUUID()}`,
+      blockchainID: fabricReceipt.blockchainID,
+      invocationType: fabricReceipt.invocationType,
+      methodName: fabricReceipt.methodName,
+      parameters: fabricReceipt.parameters,
+      timestamp: fabricReceipt.timestamp.toISOString(),
+      identity: fabricReceipt.signingCredentials.keychainRef,
+      cost: fabricReceipt.cost || 0,
+      carbonFootprint: CarbonFootPrintConstants(fabricReceipt.blockchainID),
+      latency: millisecondsLatency(fabricReceipt.timestamp),
+      revenue: fabricReceipt.revenue || 0,
+    };
+
+    this.crossChainLog.addCrossChainEvent(ccEventFromFabric);
+    this.log.info("Added Cross Chain event from FABRIC");
+    this.log.debug(`Cross-chain log: ${JSON.stringify(ccEventFromFabric)}`);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Please add a `createCrossChainEventFromEVMReceipt` to add support for EVM chains. See BUNGEE contributions from Eduardo. Code should be very similar to Besu's.